protocols/kad/src/behaviour.rs

```diff
@@ -2655,6 +2618,13 @@ where
     }
 }
 
+/// Peer Info combines a Peer ID with a set of multiaddrs that the peer is listening on.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PeerInfo {
```

Comment:
I wonder if the concept of `PeerInfo` could also be used outside of `kad`.

Close existing types:
* Kad's [`KadPeer`](https://github.com/libp2p/rust-libp2p/blob/3759e190d1e2e83f87d4be191ea7d6f747d31b6c/protocols/kad/src/protocol.rs#L85-L94)
* Gossipsub's [`PeerInfo`](https://github.com/libp2p/rust-libp2p/blob/3759e190d1e2e83f87d4be191ea7d6f747d31b6c/protocols/gossipsub/src/types.rs#L198-L203)
* Autonat protobuf [`PeerInfo`](https://github.com/libp2p/rust-libp2p/blob/3759e190d1e2e83f87d4be191ea7d6f747d31b6c/protocols/autonat/src/generated/structs.rs#L62-L65)

They have slightly different purposes, but it would be great to consolidate where possible to avoid confusion.

Reply:
yeah thanks for bringing this @guillaumemichel. It's unfortunate, but of those the one I think we could maybe get rid of would be `KadPeer` but then we'd have to deal with the state,
Regarding the other protocols `PeerInfo` like `gossipsub` after speaking with @thomaseizinger it's easier to have duplicates among protocols rather than having them depend on `libp2p-core` and deal with breaking changes

Reply:
IMO, `libp2p-core` is enough of a dependency magnet already. It needs to be split into smaller crates that have a clearer focus.

See https://github.com/libp2p/rust-libp2p/discussions/3072 for a long discussion on this. It has wide-reaching consequences :)